integration/spark/src/main/common2.3and2.4/org/apache/spark/sql/CarbonDataSourceScanHelper.scala

```diff
@@ -56,8 +56,11 @@ abstract class  CarbonDataSourceScanHelper(relation: CarbonDatasourceHadoopRelat
       .map(QueryPlan.normalizeExprId(_, output))
 
   @transient lazy val indexFilter: IndexFilter = {
-    val filter = pushedDownFilters.reduceOption(new AndExpression(_, _))
-      .map(new IndexFilter(relation.carbonTable, _, true)).orNull
+    var filter: IndexFilter = null
```

Comment:
No need to handle here. In CarbonDataSourceScan.doCanonicalize, instead of passing null, can pass Seq.empty for pushedDownFilters

Reply:
done